File: app/src/main/java/com/blighter/algoprog/Fragments/ModuleFragment.java
package com.blighter.algoprog.Fragments;

import android.os.AsyncTask;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.webkit.WebSettings;
import android.webkit.WebView;
import android.widget.Toast;

import com.blighter.algoprog.R;

import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;

import java.io.IOException;

public class ModuleFragment extends Fragment {
    String url;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_module, container, false);
        String id = null;
        Bundle bundle = getArguments();
        if (bundle != null) {
            id = bundle.getString("id");
        }
        myAsyncTask showModule = new myAsyncTask(view, id);
        showModule.execute();
        return view;
    }

    private static class myAsyncTask extends AsyncTask<Void, Void, Document> {
        String basicUrl = "https://algoprog.ru/material/";
        View cont;
        Boolean allIsOk = true;

        myAsyncTask(View context, String id) {
            cont = context;
            basicUrl = basicUrl + id;
        }

        @Override
        protected Document doInBackground(Void... voids) {
            Document doc = null;
            try {
                doc = Jsoup.connect(basicUrl).get();
                doc.getElementsByClass("navbar navbar-default navbar-fixed-top").remove();
                doc.getElementsByClass("_client_components_Sceleton__footer").remove();
                doc.getElementsByClass("breadcrumb").remove();
            } catch (IOException e) {
                allIsOk = false;
            }

            return doc;
        }

        @Override
        protected void onPostExecute(Document document) {
            super.onPostExecute(document);
            if (allIsOk) {
                final WebView browser = cont.findViewById(R.id.browser);
                browser.loadDataWithBaseURL(basicUrl, document.toString(), "text/html", "utf-8", "");
                browser.getSettings().setCacheMode(WebSettings.LOAD_CACHE_ELSE_NETWORK);
            } else
                Toast.makeText(cont.getContext(), "Нет подключения к интернету", Toast.LENGTH_SHORT).show();
        }
    }
}

File: app/src/main/java/com/blighter/algoprog/fragments/ModuleFragment.kt
package com.blighter.algoprog.fragments

import android.content.Context
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.webkit.WebSettings
import android.webkit.WebView
import android.widget.Toast
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentActivity
import com.blighter.algoprog.R
import com.blighter.algoprog.api.Errors
import com.blighter.algoprog.api.scopeNetwork
import com.blighter.algoprog.utils.UrlInterceptor
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import kotlinx.coroutines.withContext
import org.jsoup.Jsoup
import org.jsoup.nodes.Document

//this fragment shows any module from Algoprog
class ModuleFragment : Fragment() {
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_module, container, false)
        val url: String?
        val bundle = arguments
        //url check, that we got from intent and start setModule()
        if (bundle != null) {
            url = bundle.getString("url")
            setModule(requireContext(), url, view.findViewById(R.id.browser))
        } else {
            Toast.makeText(context, "Что-то пошло не так", Toast.LENGTH_LONG).show()
            requireActivity().onBackPressed()
        }
        return view
    }

    private fun setModule(context: Context, url: String?, browser: WebView) {
        scopeNetwork.launch {
            var doc: Document? = null
            var error = false
            try {
                doc = Jsoup.connect(url).get()
            } catch (e: Exception) {
                error = true
            }
            withContext(Dispatchers.Main) {
                when {
                    error -> {
                        Toast.makeText(context, Errors.NoInternet.warn, Toast.LENGTH_SHORT).show()
                        (context as FragmentActivity).onBackPressed()
                    }
                    doc == null -> {
                        Toast.makeText(context, Errors.Unknown.warn, Toast.LENGTH_LONG).show()
                    }
                    else -> {
                        //delete all useless html code
                        doc.getElementsByClass("navbar navbar-default navbar-fixed-top").remove()
                        doc.getElementsByClass("_client_components_Sceleton__footer").remove()
                        doc.getElementsByClass("breadcrumb").remove()
                        doc.getElementsByClass("_2ypNosdSxDScDQXykk62PR").remove()
                        //setting our custom WebViewClient ( class from utils)
                        browser.webViewClient = UrlInterceptor(context)
                        browser.settings.cacheMode = WebSettings.LOAD_CACHE_ELSE_NETWORK
                        browser.settings.textZoom = 115
                        browser.loadDataWithBaseURL(url, doc.toString(), "text/html", "utf-8", "")
                    }
                }
            }
        }
    }
}